compiler-core/src/ast/typed.rs

```diff
@@ -47,6 +47,9 @@ pub enum TypedExpr {
 
     Fn {
         location: SrcSpan,
+        /// The location of the function head, if this function was defined
+        /// using anonymous function syntax, as opposed to a capture.
+        head_location: Option<SrcSpan>,
```

Comment:
This now duplicates the `is_capture` property. Let's replace both with a `kind: FunctionLiteralKind::{Capture, Anonymous}`, or something like that. The variant for not-a-capture would hold the location of the head.

Reply:
Sounds good! I didn't really like this method anyway